homeassistant/components/filter/sensor.py

```diff
@@ -432,7 +434,9 @@ def __init__(
         else:
             self.states = deque(maxlen=0)
             self.window_unit = WINDOW_SIZE_UNIT_TIME
-        self.filter_precision = precision
+        self.filter_precision = (
+            precision if precision is not None else self._default_precision
+        )
```

Comment:
I think the type hint on line 422 is wrong.

Reply:
you are right! (fixed)